.github/workflows/build-publish.yml

```diff
@@ -0,0 +1,34 @@
+name: 'Build, Lint, Test and Deploy if master'
+
+on: push
+
+jobs:
+  build:
+    name: Build and release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Install dependencies
+        run: npm i
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+
+      - run: npm run test
+      - run: npm run lint
+
+      - name: Dryrun release
+        run: npm run release:dryrun
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Semantic release
+        run: npm run release
+        if: github.ref == 'refs/heads/master'
```

Comment:
If you don't like the if statement here, you can define semantic-release config in your package.json, for example I've set master-only here:
https://github.com/telus/poc-github-npm-package/blob/master/package.json#L49-L56

Reply:
Rather keep it on the workflow, so that is less prone to accidental modification